services/movie_session.py

```diff
@@ -0,0 +1,57 @@
+from datetime import datetime
+
+from django.db.models import QuerySet
+
+from db.models import MovieSession, Movie, CinemaHall
+
+
+def create_movie_session(
+        movie_show_time: datetime,
+        movie_id: int,
+        cinema_hall_id: int,
+) -> MovieSession:
+    new_movie_session = MovieSession.objects.create(
+        show_time=movie_show_time,
+        cinema_hall=CinemaHall.objects.get(id=cinema_hall_id),
+        movie=Movie.objects.get(id=movie_id)
+    )
+    return new_movie_session
+
+
+def get_movies_sessions(session_date: str = None) -> QuerySet:
+    if session_date:
+        target_time = datetime.strptime(session_date, "%Y-%m-%d").date()
+        return MovieSession.objects.filter(show_time__date=target_time)
+    return MovieSession.objects.all()
+
+
+def get_movie_session_by_id(movie_session_id: int) -> MovieSession:
+    return MovieSession.objects.get(id=movie_session_id)
+
+
+def update_movie_session(
+        session_id: int,
+        show_time: datetime = None,
+        movie_id: int = None,
+        cinema_hall_id: int = None
+) -> None:
+    movie_session = MovieSession.objects.get(id=session_id)
+
+    if show_time:
+        movie_session.show_time = show_time
+
+    if cinema_hall_id:
+        cinema_hall_inst = CinemaHall.objects.filter(id=cinema_hall_id)
+        if cinema_hall_inst.exists():
+            movie_session.cinema_hall = cinema_hall_inst.first()
+
+    if movie_id:
```

Comment:
Can you refactor this so that you will not need to get movie session if none of the update parameters has been passed?

Reply:
Excuse me, can you please explain this note to me as I am getting the movie session only if some parameters are passed

Reply:
Try to implement this:
1. Create an empty dict
2. Check if any of the optional arguments are passed to this function. In case if there are - you will add them to the dict.
3.  In the end check if your dict has some values to update and if yes you will call update on queryset, and if there are not any you don't do anything.

Reply:
almost done it:
remove line 39 where you get movie session and just put
`MovieSession.objects.filter(pk=session_id).update(**updating_data)` 
under `if updating_data:`